test/functional/wallet_resendwallettransactions.py

```diff
@@ -32,45 +31,45 @@ def skip_test_if_missing_module(self):
+        node.settxfee(relayfee)
+        txhsh = node.sendtoaddress(node.getnewaddress(), 1)
+
+        assert txhsh in node.getrawmempool()
+
+        # bump mocktime so the transaction should expire
+        # add an extra hour for good measure
+        two_weeks_in_seconds = 60 * 60 * 24 * 14
+        mocktime = int(time.time()) + two_weeks_in_seconds + 60 * 60
+        node.setmocktime(mocktime)
+
+        # making a new transaction invokes ATMP which expires old txns
+        node.sendtoaddress(node.getnewaddress(), 1)
+
+        # confirm txn is no longer in mempool
+        self.log.info("Confirm transaction is no longer in mempool")
+        assert txhsh not in node.getrawmempool()
+
+        # bumptime so ResendWalletTransactions triggers
+        # we resend once / day, so bump 25 hours just to be sure
+        # we don't resubmit the first time, so we bump mocktime
+        # twice so the resend occurs the second time around
+        one_day_in_seconds = 60 * 60 * 25
+        node.setmocktime(mocktime + one_day_in_seconds)
 
-        # Wallet rebroadcast is first scheduled 1 sec after startup (see
-        # nNextResend in ResendWalletTransactions()). Sleep for just over a
-        # second to be certain that it has been called before the first
-        # setmocktime call below.
         time.sleep(1.1)
 
-        # Can take a few seconds due to transaction trickling
-        wait_until(lambda: node.p2p.tx_invs_received[txid] >= 1, lock=mininode_lock)
-
-        # Add a second peer since txs aren't rebroadcast to the same peer (see filterInventoryKnown)
-        node.add_p2p_connection(P2PStoreTxInvs())
-
-        self.log.info("Create a block")
-        # Create and submit a block without the transaction.
-        # Transactions are only rebroadcast if there has been a block at least five minutes
-        # after the last time we tried to broadcast. Use mocktime and give an extra minute to be sure.
-        block_time = int(time.time()) + 6 * 60
-        node.setmocktime(block_time)
-        block = create_block(int(node.getbestblockhash(), 16), create_coinbase(node.getblockcount() + 1), block_time)
-        block.rehash()
-        block.solve()
-        node.submitblock(ToHex(block))
-
-        # Transaction should not be rebroadcast
-        node.p2ps[1].sync_with_ping()
-        assert_equal(node.p2ps[1].tx_invs_received[txid], 0)
-
-        self.log.info("Transaction should be rebroadcast after 30 minutes")
-        # Use mocktime and give an extra 5 minutes to be sure.
-        rebroadcast_time = int(time.time()) + 41 * 60
-        node.setmocktime(rebroadcast_time)
-        wait_until(lambda: node.p2ps[1].tx_invs_received[txid] >= 1, lock=mininode_lock)
+        node.setmocktime(mocktime + 2 * one_day_in_seconds)
+
+        # confirm that its back in the mempool
+        self.log.info("Transaction should be resubmitted to mempool")
+        wait_until(lambda: txhsh in node.getrawmempool(), timeout=30)
```

Comment:
This goes into timeout for me, causing `wallet_resendwallettransactions` to fail when I run the entire test suite (same as in AppVeyor build here) but the test succeeds if I run it in isolation. If you can’t reproduce this I could look deeper into why it fails.

Reply:
I think that this is due to a bug with the time: `RESEND_TXS_FREQUENCY` is in microseconds. It is added to `GetTime()`  which is in seconds, so an actual resend doesn't ever happen in test. 
Yet the test sometimes succeeds because there is an initial resend (`nNextResend` is initialized to 0) which in some runs happens after the mocktime is set to two weeks and the tx has been expired (test passes), in some runs happens earlier (test fails).

Reply:
On my other computer it also sometimes failed in the single run, which made testing easier. I found this the bug adding some debug statements in `ResendWalletTransactions`. 

Fixing the times might not solve this completely, because the initial rebroadcast after startup also happened for me in a few runs right before `assert txhsh not in node.getrawmempool()`, which then fails. Adding a sleep before you jump ahead in time could help.